File: TestProject/Assets/Plugins/PlatformSDK/AndroidCeliaOversea/ElvaHelper.java
package celia.sdk;
import android.content.Intent;
import android.net.Uri;

import com.ljoy.chatbot.sdk.ELvaChatServiceSdk;
import org.json.JSONException;
import org.json.JSONObject;

import java.text.MessageFormat;
import java.util.HashMap;

public class ElvaHelper {
    CeliaActivity mainActivity;
    String playerUid = "userId123";
    String playerName = "userName123";
    String serverID = "serverId123";
    String gameName = "Girl for the Throne TW";
    String PlayershowConversationFlag = "1";
    public ElvaHelper(CeliaActivity activity)
    {
        mainActivity = activity;
        Init();
    }
    public void Init(){
//        ELvaChatServiceSdk.setName(gameName);
//        ELvaChatServiceSdk.setUserId(playerUid);
//        ELvaChatServiceSdk.setUserName(playerName);
//        ELvaChatServiceSdk.setServerId(serverID);
//        ELvaChatServiceSdk.setSDKLanguage("zh_CN");
        setInitCallback();
        ELvaChatServiceSdk.init(mainActivity,Constant.Elva_AppKey,Constant.Elva_Domain,Constant.Elva_AppId);
        String languageAlias = "zh_TW";
        ELvaChatServiceSdk.setSDKLanguage(languageAlias);

    }
    // 在调用初始化init方法之前，设置初始化回调函数
    public void setInitCallback() {
        ELvaChatServiceSdk.setOnInitializedCallback(new ELvaChatServiceSdk.OnInitializationCallback() {
            @Override
            public void onInitialized() {
                mainActivity.ShowLog("---ElvaHelper---");
            }
        });
    }
    public void Show(String jsonStr){
        try{
            JSONObject jsonObject = new JSONObject(jsonStr);
            playerName = jsonObject.getString("playerName");
            playerUid = jsonObject.getString("playerUid");
            serverID = jsonObject.getString("ServerID");
            PlayershowConversationFlag = jsonObject.getString("PlayershowConversationFlag");
            ELvaChatServiceSdk.setUserId(playerUid);
            ELvaChatServiceSdk.setUserName(playerName);
            int type = Integer.parseInt(jsonObject.getString("Type"));
            if (type == 1){
                ShowElva();
            }else if (type == 2){
                showFAQs();
            }
            else if (type == 3){
                showElvaOP();
            }
            else if (type == 4){
                showConversation();
}            else if (type == 5){
                showSuggestWindow(jsonObject.getString("formID"));
            }else {
                ShowElva();
            }
        }catch (JSONException e){
            e.printStackTrace();
        }
    }
    //1.调用showElva接口，启动机器人客服聊天界面
    public void ShowElva(){
        ELvaChatServiceSdk.showElva(playerName, playerUid, serverID, PlayershowConversationFlag);
    }
    //2.调用showConversation方法，启动人工客服界面
    public void showConversation(){
        ELvaChatServiceSdk.showConversation(playerUid, serverID);
    }
    //3.调用showElvaOP方法，启动运营模块界面
    public void showElvaOP(){
        ELvaChatServiceSdk.showElvaOP(playerName, playerUid, serverID, PlayershowConversationFlag);
    }
    //4.展示FAQ列表，调用showFAQs 方法
    public void showFAQs(){
        HashMap<String, Object> config = new HashMap();
        HashMap<String, Object> map = new HashMap();
        // "elva-custom-metadata" 是key值 不可以变
        config.put("elva-custom-metadata", map);
        config.put("showContactButtonFlag", "1");
        config.put("directConversation", "1");
        ELvaChatServiceSdk.showFAQs(config);
//        ELvaChatServiceSdk.showFAQSection("1234",config);
    }
    //5.外部打开反馈表单
    public void showSuggestWindow(String formId){
        String s = MessageFormat.format("https://aihelp.net/questionnaire/#/?formId={0}&appId={1}&uid={2}&userName={3}",formId,
                Constant.Elva_AppId,playerUid,playerName);
        Uri uri=Uri.parse("https://aihelp.net/questionnaire/#/?formId=59a639c15027487ba447551b1f8d16cb&appId=elextech_platform_c72b5e80-ec84-45d7-9fe6-212c0800c423&uid=123&userName=456");
        Intent intent=new Intent(Intent.ACTION_VIEW,uri);
        mainActivity.startActivity(intent);
    }
}

File: TestProject/Assets/Plugins/PlatformSDK/IOSCeliaOversea/ElvaHelper.m
//
//  ElvaHelper.m
//  Unity-iPhone
//
//  Created by mini on 8/5/20.
//

#import "ElvaHelper.h"

@implementation ElvaHelper{
    NSString *AIHelpAppID;
    NSString *playerName;
    NSString *playerUid;
    NSString *serverId;
    NSString *playerParseId;
    NSString *showConversationFlag;
    id IOSBridgeHelper;
}
static ElvaHelper *ElvaHelperIns = nil;
+(ElvaHelper*)sharedInstance{
    if (ElvaHelperIns == nil) {
        ElvaHelperIns = [ElvaHelper new];
    }
    return ElvaHelperIns;
}
-(void)InitSDK:(id)delegate{
    IOSBridgeHelper = delegate;
    NSDictionary *infoDictionary = [[NSBundle mainBundle] infoDictionary];
    AIHelpAppID = [infoDictionary objectForKey:@"AIHelpAppID"];
    NSString *AIHelpAppKey = [infoDictionary objectForKey:@"AIHelpAppKey"];
    NSString *AIHelpDomain = [infoDictionary objectForKey:@"AIHelpDomain"];
    [ECServiceSdk setSDKInterfaceOrientationMask:UIInterfaceOrientationMaskPortrait];
    [ECServiceSdk init:AIHelpAppKey Domain:AIHelpDomain AppId:AIHelpAppID];
    playerParseId = @"";
//    [ECServiceSdk setName:@"少女的王座"];
//    [ECServiceSdk setServerId:@"ServerId-123456987456"];
    [ECServiceSdk setSDKLanguage:@"zh_TW"];
//    [ECServiceSdk setSDKLanguage:@"zh_en"];
//    [ECServiceSdk setSDKLanguage:@"zh_CN"];
    NSLog(@"--ElvaHelper---InitSDK---");
}
#pragma mark --入口
-(void)show:(const char*) jsonString{
    NSString *jsonNSString = [NSString stringWithUTF8String:jsonString];
    NSData *data = [jsonNSString dataUsingEncoding:NSUTF8StringEncoding];
    NSMutableDictionary *dict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
    NSLog(@" ---ios ElvaHelper--show-: %@", dict);
    playerName = [dict valueForKey:@"playerName"];
    playerUid = [dict valueForKey:@"playerUid"];
    serverId = [dict valueForKey:@"ServerID"];

    showConversationFlag = [dict valueForKey:@"PlayershowConversationFlag"];
    
    int type = [[dict valueForKey:@"Type"] intValue];

    [ECServiceSdk setUserName:playerName];
    [ECServiceSdk setUserId:playerUid];

    if (type == 1) {
        [self showElva];
    }else if (type == 2){
        [self showFAQs];
    }
    else if (type == 3){
        [self showElvaOP];
    }
    else if (type == 4){
        [self showConversation];
    }
    else if (type == 5){
        [self showSuggestWindow:[dict valueForKey:@"formID"]];
    }
    else{
        [self showElva];
    }

}
#pragma mark --智能客服主界面启动，调用 showElva 方法，启动机器人界面
-(void)showElva{
    [ECServiceSdk showElva:playerName
                 PlayerUid:playerUid
                  ServerId:serverId
             PlayerParseId:playerParseId
    PlayershowConversationFlag:showConversationFlag];
}
#pragma mark --展示FAQ列表，调用 showFAQs 方法(必须确保设置用户名称信息 setUserName 和设置用户唯一id信息 setUserId 已经调用)
-(void)showFAQs{
    NSMutableDictionary *customData = [NSMutableDictionary dictionary];//定义自定义参数容器
    NSMutableDictionary *config = [NSMutableDictionary dictionary]; //定义config参数容器
    [config setObject:@"1" forKey:@"showContactButtonFlag"];
    [config setObject:@"1" forKey:@"directConversation"];
    [config setObject:customData forKey:@"elva-custom-metadata"]; //将customData存入容器
    [ECServiceSdk showFAQs:config];
    //[ECServiceSdk showFAQSection:@"158455" Config:config];//展示相关部分FAQ SectionID需要后台配置
}
#pragma mark --运营主界面启动
-(void)showElvaOP{
    [ECServiceSdk showElvaOP:playerName
                   PlayerUid:playerUid
                    ServerId:serverId
               PlayerParseId:playerParseId
  PlayershowConversationFlag:showConversationFlag
                      Config:nil];
}
#pragma mark --直接进行人工客服聊天，调用 showConversation 方法(必须确保设置用户名称信息 setUserName 已经调用)
-(void)showConversation{
    [ECServiceSdk showConversation:playerUid ServerId:serverId];
}
#pragma mark --外部打开反馈表单
-(void)showSuggestWindow:(NSString *)formID{
    NSString *name = [playerName stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
    NSString *urlStr = [NSString stringWithFormat:@"https://aihelp.net/questionnaire/#/?formId=%@&appId=%@&uid=%@&userName=%@",formID,AIHelpAppID,playerUid,name];

    NSURL *url = [NSURL URLWithString: urlStr];
    NSLog(@"-ios----showSuggestWindow---url----%@",url);
    if ([[UIApplication sharedApplication] canOpenURL:url]) {
        if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 10.0) {
                //设备系统为IOS 10.0或者以上的
                [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
            }else{
                //设备系统为IOS 10.0以下的
                [[UIApplication sharedApplication] openURL:url];
            }
    }


}
@end
